### ISSUE TITLE
"Unexpected end of file from server" error on sbt codeArtifactPublish

### ISSUE THREAD

**Issue by tuzonghua** [bbstilson/sbt-codeartifact#9]
Followed the readme to try and publish to CodeArtifact. Getting the error `(codeArtifactPublish) java.net.SocketException: Unexpected end of file from server` when running the command `sbt codeArtifactPublish`.

My build.sbt:
```
lazy val root = (project in file(".")).
  settings(
    name := "testPackage",
    organization := "org.myorg",
    version := "0.1.0-SNAPSHOT",
    versionScheme := Some("semver-spec"),
    scalaVersion := "2.13.6",
    codeArtifactUrl := "https://test-domain-1234567890.d.codeartifact.us-east-1.amazonaws.com/maven/test-repo/"
  )
```

Console output and stacktrace when running `sbt codeArtifactPublish`:
```
$ sbt codeArtifactPublish
WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by sbt.TrapExit$ (file:/Users/tuzonghua/.sbt/boot/scala-2.12.14/org.scala-sbt/sbt/1.5.5/run_2.12-1.5.5.jar)
WARNING: Please consider reporting this to the maintainers of sbt.TrapExit$
WARNING: System::setSecurityManager will be removed in a future release
[info] welcome to sbt 1.5.5 (Homebrew Java 17)
[info] loading settings for project scala_module-build from plugins.sbt ...
[info] loading project definition from /Users/tuzonghua/src/package-poc/scala_module/project
[info] loading settings for project root from build.sbt ...
[info] set current project to testpackage (in build file:/Users/tuzonghua/src/package-poc/scala_module/)
[info] Main Scala API documentation to /Users/tuzonghua/src/package-poc/scala_module/target/scala-2.13/api...
[info] Wrote /Users/tuzonghua/src/package-poc/scala_module/target/scala-2.13/testpackage_2.13-0.1.0-SNAPSHOT.pom
[info] Main Scala API documentation successful.
[info] Uploading org/myorg/testpackage_2.13/0.1.0-SNAPSHOT/testpackage_2.13-0.1.0-SNAPSHOT.pom
[info] Uploading org/myorg/testpackage_2.13/0.1.0-SNAPSHOT/testpackage_2.13-0.1.0-SNAPSHOT.jar
[info] Uploading org/myorg/testpackage_2.13/0.1.0-SNAPSHOT/testpackage_2.13-0.1.0-SNAPSHOT-sources.jar
[info] Uploading org/myorg/testpackage_2.13/0.1.0-SNAPSHOT/testpackage_2.13-0.1.0-SNAPSHOT-javadoc.jar
[info] Uploading org/myorg/testpackage_2.13/maven-metadata.xml
[error] java.net.SocketException: Unexpected end of file from server
[error]   at java.base/sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:903)
[error]   at java.base/sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:726)
[error]   at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1688)
[error]   at java.base/sun.net.www.protocol.http.HttpURLConnection$9.run(HttpURLConnection.java:1581)
[error]   at java.base/sun.net.www.protocol.http.HttpURLConnection$9.run(HttpURLConnection.java:1579)
[error]   at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
[error]   at java.base/java.security.AccessController.doPrivilegedWithCombiner(AccessController.java:962)
[error]   at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1578)
[error]   at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:529)
[error]   at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:308)
[error]   at requests.Requester$$anon$1.liftedTree1$1(Requester.scala:268)
[error]   at requests.Requester$$anon$1.readBytesThrough(Requester.scala:267)
[error]   at geny.Readable.writeBytesTo(Writable.scala:93)
[error]   at geny.Readable.writeBytesTo$(Writable.scala:93)
[error]   at requests.Requester$$anon$1.writeBytesTo(Requester.scala:165)
[error]   at requests.Requester.apply(Requester.scala:114)
[error]   at codeartifact.CodeArtifactApi.upload(CodeArtifactApi.scala:21)
[error]   at codeartifact.CodeArtifactPlugin$.$anonfun$publish0$6(CodeArtifactPlugin.scala:102)
[error]   at scala.collection.immutable.List.map(List.scala:297)
[error]   at codeartifact.CodeArtifactPlugin$.$anonfun$publish0$1(CodeArtifactPlugin.scala:100)
[error]   at codeartifact.CodeArtifactPlugin$.$anonfun$publish0$1$adapted(CodeArtifactPlugin.scala:72)
[error]   at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]   at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error]   at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error]   at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error]   at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error]   at sbt.Execute.work(Execute.scala:291)
[error]   at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error]   at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]   at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error]   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]   at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error]   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error]   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error]   at java.base/java.lang.Thread.run(Thread.java:833)
[error] (codeArtifactPublish) java.net.SocketException: Unexpected end of file from server
[error] Total time: 2 s, completed Sep 28, 2021, 4:25:16 PM
```

**Comment by bbstilson** [comment]
I'm sad to say this is likely related to this issue: https://github.com/bbstilson/sbt-codeartifact/issues/8. Specifically, this plugin does not support SNAPSHOT jars at the moment. I've made some progress in addressing this shortcoming, but I just started a new job and so haven't had a ton of time (or energy) to work on it. 

My suggestion at the moment is to remove the `-SNAPSHOT` from your version and see if that fixes it.
